arangod/Aql/SingleRemoteModificationExecutor.cpp

```diff
@@ -198,7 +213,8 @@ auto SingleRemoteModificationExecutor<
     }
   }
 
-  stats.incrWritesExecuted(possibleWrites);
+  stats.incrWritesExecuted(writesExecuted);
+  stats.incrWritesIgnored(writesIgnored);
   stats.incrScannedIndex();
```

Comment:
The `incrScannedIndex()` is not appropriate for write queries.
`incrScannedIndex()` should only be called if the optimizer rule is applied for a read query. Can you fix this and also add a test for this? Thanks!